luigi/task.py

```diff
@@ -61,6 +67,19 @@ class Task2(luigi.Task):
     Register._default_namespace = namespace
 
 
+def task_id_str(task_family, params):
```

Comment:
Can this either be made private (start with `_`) or be given a docstring?

This refactoring could happen in  a seperate PR to make this PR smaller. (If you have the time to refactor that)